internal/cluster/distribution/eks/ekscluster/nodepools/wait_for_asg.go

```diff
@@ -80,7 +81,21 @@ func WaitForASGToBeFulfilled(
 					return nil
 				}
 			} else {
-				return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s", waitAttempts, waitInterval)
+				asGroup, err := m.GetAutoscalingGroupByStackName(asgName)
+				if err != nil {
+					return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s, could not get ASG in stack: %s", waitAttempts, waitInterval, asgName)
+				}
+				lastActivity, _ := m.LastAutoscalingActivity(asGroup.AutoScalingGroupName)
+				if lastActivity != nil {
+					return errors.Errorf(
+						"waiting for ASG to be fulfilled timed out after %d x %s, status: %s, cause: %s, description: %s",
+						waitAttempts,
+						waitInterval,
+						aws.StringValue(lastActivity.StatusCode),
+						aws.StringValue(lastActivity.Cause),
+						aws.StringValue(lastActivity.Description),
+					)
+				}
```

Comment:
**Question**: is there no error in the spot price timeout case in the `GetAutoscalingGroupByStackName()` function?

Reply:
There is no error but it can be inferred there is a problem by checking the length of the Instances slice (if the instances cannot come up this is empty).